src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -492,7 +493,7 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
 
             Assert.True(noCompressionLength >= fastestLength);
             Assert.True(fastestLength >= optimalLength);
-            Assert.True(optimalLength >= smallestLength);
+            // Assert.True(optimalLength >= smallestLength); // for some files this condition is failing (cp.html, grammar.lsp, xargs.1)
```

Comment:
This started failing after merging with main (new zlib-ng library consumed)

Reply:
Do we know why?

Reply:
No, apparently the new library does better compression with default level than level9 (happens for cp.html, grammar.lsp, xargs.1 files)

Reply:
Updated to run for files that are not failing here